opds.py

```diff
@@ -703,6 +703,12 @@ def page(cls, _db, title, url, lane, annotator,
             search_engine=search_engine, debug=search_debug
         )
 
+        if not isinstance(works, list):
+            # It's possible that works() returned a database query or
+            # other generator-like object, but at this point we want
+            # an actual list of Work objects.
+            works = [x for x in works]
+
```

Comment:
And this makes OPDS feed generation work for both database and search engine lanes. Earlier I changed it from only working for database to only working for search engine.